### ISSUE TITLE
add_hparams() NoneType error

### ISSUE THREAD

**Issue by gth879w** [pytorch/pytorch#29775]
## Issue description
`add_hparams()` in `torch.utils.tensorboard.writer` produced the following error
```
python3.7/site-packages/torch/utils/tensorboard/writer.py", line 294, in add_hparams
    with SummaryWriter(log_dir=os.path.join(self.file_writer.get_logdir(), str(time.time()))) as w_hp:
AttributeError: 'NoneType' object has no attribute 'get_logdir'
```
Other methods such as `add_scalar()` and `add_histogram()` use `self._get_file_writer()` instead of `self.file_writer` directly. This ensures `self.file_writer` is initialized properly. I think `add_hparams()` should do the same.

## Code example

I don't have a self contained minimal example, but I think the issue is apparent.

## System Info
- PyTorch or Caffe2: PyTorch
- How you installed PyTorch (conda, pip, source): conda
- Build command you used (if compiling from source):
- OS: WSL
- PyTorch version: 1.3.1
- Python version: 3.7
- CUDA/cuDNN version: None
- GPU models and configuration: None
- GCC version (if compiling from source): (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
- CMake version: N/A
- Versions of any other relevant libraries:

**Comment by vincentqb** [comment]
Have you tried compiling the code from source with this change? We'd welcome a pull request for a change like this. Thank you!

**Comment by natalialunova** [comment]
Closed by PR #30286

**Comment by aliechoes** [comment]
I also faced the same error in a situation.
So if I do not run the tensorboard in parallel (tensorboard is not listening) eveything works. But as soon as tensorboard is listening, add_hparams() gives this error:
```
AttributeError: 'NoneType' object has no attribute 'get_logdir'
```
